src/middlewared/middlewared/webhooks/cluster_events.py

```diff
@@ -0,0 +1,81 @@
+from middlewared.plugins.cluster_linux.utils import CTDBConfig
+
+from aiohttp import web
+
+
+class ClusterEventsApplication(object):
+
+    def __init__(self, middleware):
+        self.middleware = middleware
+        self.ctdb_shared_vol = CTDBConfig.CTDB_VOL_NAME.value
+
+    async def process_event(self, data):
+
+        event = data.get('event', None)
+        msg = data.get('message', None)
+
+        umount_it = mount_it = False
+        if event and msg:
+            if event == 'VOLUME_START':
+                if msg.get('name', '') == self.ctdb_shared_vol:
+                    mount_it = True
+
+            if event == 'AFR_SUBVOL_UP':
+                if msg.get('subvol', '') in self.ctdb_shared_vol:
+                    mount_it = True
+
+            if event == 'VOLUME_STOP':
+                if msg.get('name', '') == self.ctdb_shared_vol:
+                    umount_it = True
+
+            if event == 'AFR_SUBVOLS_DOWN':
+                if msg.get('subvol', '') in self.ctdb_shared_vol:
+                    umount_it = True
+
+        if mount_it:
+            mount = await(
+                await self.middleware.call('ctdb.shared.volume.mount')
+            ).wait()
```

Comment:
Is it OK to block process_event like we are waiting it to mount/umount?

Reply:
Shouldn't matter, but the `volume.{mount/umount}` methods share the same lock so we should be protected anyways.